packages/alpinejs/src/directives/x-model.js

```diff
@@ -169,6 +187,18 @@ function safeParseNumber(rawValue) {
     return isNumeric(number) ? number : rawValue
 }
 
+function safeParseBoolean(rawValue) {
+    if ([1, '1', 'true', true].includes(rawValue)) {
+        return true
+    }
+
+    if ([0, '0', 'false', false].includes(rawValue)) {
+        return false
```

Comment:
I was thinking "How could we do this with only one comparison?" and my brain went to 

```js
Boolean(JSON.parse(rawValue.toString()))
```

But surely that would stupid right?

Reply:
I think that does not work, because if the rawValue is not 1/0/true/false, then we want to just return that raw value. With that code, I don't think that is possible 🤔

Reply:
Yes, I do see that. To me though, the use case of this modifier seems fundamentally contradictory to ever having it be any other values. I'd rather have it fail early (at a json parsing error) than continue to have the bad value end up somewhere else somehow.

But that's an opinionated design.

Reply:
The `x-model.number` works like that as well so I thought it might be best to let it behave in the same way
https://github.com/alpinejs/alpine/blob/6142a83956685b4ddf67d95cf70a46fa3efe5fdf/packages/alpinejs/src/directives/x-model.js#L166-L170

Reply:
Makes sense 👍